src/Kernel32.Shared/Kernel32Extensions.cs

```diff
@@ -55,5 +63,30 @@ public static void ThrowOnError(this NTSTATUS status)
                 throw new NTStatusException(status);
             }
         }
+
+        /// <summary>
+        /// Throws an exception with localized message when an error occurs.
+        /// </summary>
+        /// <param name="errorCode">The result of the P/Invoke call.</param>
+        /// <exception cref="Win32Exception">If <paramref name="errorCode"/> is not <see cref="Win32ErrorCode.ERROR_SUCCESS"/></exception>
+        public static void ThrowOnLocalizedError(this Win32ErrorCode errorCode)
```

Comment:
Since this is a pure helper method (no native equivalent), it seems quite reasonable to have it localize the message in the thrown exception by default. So perhaps we can always localize in these methods, and not have the extra 'localized' versions.